internal/controllers/reconciliation/label_selector_test.go

```diff
@@ -0,0 +1,109 @@
+package reconciliation
+
+import (
+	"testing"
+
+	"github.com/Azure/eno/internal/resource"
+	"github.com/stretchr/testify/assert"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+func TestResourceLabelSelectorFilter(t *testing.T) {
```

Comment:
Is this sufficient test coverage? Consider adding an integration test if needed.